GH-3105: Spring integration stream SFTP: pattern filter not wrking

lukasGemela opened the pull request:
https://github.com/spring-projects/spring-integration/issues/3105

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

artembilan commented on spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java:
```diff
@@ -190,9 +190,10 @@ protected Object doReceive() {
 					}
 					else {
-						file = null;
+						break;
```
Grate!

I think we don't need an `if (file != null)` any more after this.

lukasGemela replied:
Sure thing will do tomorrow

lukasGemela replied:
@artembilan test added

artembilan reviewed (changes requested):
Please, add your name to the `@author` list.
And also it would be great to have some test to cover this functionality.

I guess if we would have had such a test before we wouldn't have this bug at all 😢 

Thank you for your contribution anyway!

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

code review comments

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

added test

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

fixed compilation issue

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

fixed validation issue

Commit:
fixes GH-3105: Spring integration stream SFTP: pattern filter not working

https://github.com/spring-projects/spring-integration/issues/3105

fixed validation issue

artembilan reviewed (approved):
Grate!

Pulling locally for final review and merge after code clean up.

👍

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/1820c05c26ad625d449d49b867a72f8d1cf97571.

@lukasGemela ,

Thank you for contribution; looking forward for more!
